packages/devtools_app/lib/src/memory/memory_android_chart.dart

```diff
@@ -15,10 +15,88 @@ import '../theme.dart';
 import 'memory_controller.dart';
 import 'memory_timeline.dart';
 
+class AndroidChartController extends ChartController {
+  AndroidChartController(this._memoryController)
+      : super(
+          displayTopLine: false,
+          name: 'Android',
+        );
+
+  final MemoryController _memoryController;
+
+  // TODO(terry): Only load max visible data collected, when pruning of data
+  //              charted is added.
+  /// Preload any existing data collected but not in the chart.
+  @override
+  void setupData() {
+    final chartDataLength = timestampsSize;
+    final dataLength = _memoryController.memoryTimeline.data.length;
+
+    final dataRange = _memoryController.memoryTimeline.data.getRange(
+      chartDataLength,
+      dataLength,
+    );
+
+    dataRange.forEach(addSampleToChart);
+  }
+
+  /// Loads all heap samples (live data or offline).
+  void addSampleToChart(HeapSample sample) {
```

Comment:
nit: is the `ToChart` needed as this is thechart class. maybe just call this `addSample`?

Reply:
Done.